core/trino-main/src/main/java/io/trino/operator/hash/IsolatedHashTableFactory.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.hash;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.airlift.bytecode.DynamicClassLoader;
+import io.trino.operator.GroupByIdBlock;
+import io.trino.operator.HashGenerator;
+import io.trino.operator.Work;
+import io.trino.operator.hash.HashTableDataGroupByHash.AddPageWork;
+import io.trino.operator.hash.HashTableDataGroupByHash.GetGroupIdsWork;
+import io.trino.operator.hash.fixedwidth.FixedWidthEntryStructure;
+import io.trino.operator.hash.fixedwidth.GenericFixedWidthHashTableRowHandler;
+import io.trino.operator.hash.fixedwidth.GenericFixedWidthHashTableValuesAppender;
+import io.trino.operator.hash.fixedwidth.gen.FixedWidthHashTableRowHandler1Channels;
+import io.trino.operator.hash.fixedwidth.gen.FixedWidthHashTableValuesAppender1Channels;
+import io.trino.spi.block.Block;
+import io.trino.sql.gen.IsolatedClass;
+
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class IsolatedHashTableFactory
+{
+    public static final int MAX_SUPPORTED_CHANNELS = 20;
```

Comment:
I don't think 20 is needed. I've never seen a group by using more than a few columns.

Reply:
Unless we want to generate it with bytecode generator.